src/extractor/guidance/segregated_intersection_classification.cpp

```diff
@@ -145,18 +55,124 @@ std::unordered_set<EdgeID> findSegregatedNodes(const NodeBasedGraphFactory &fact
+            coordExtractor.GetCoordinateCloseToTurn(intersection_node, to_edge_id, false, to_node);
+        auto const node_from = coordExtractor.GetCoordinateCloseToTurn(
+            intersection_node, from_edge_id_outgoing, false, from_node);
+        return util::coordinate_calculation::computeAngle(
+            node_from, coordinates[intersection_node], node_to);
+    };
+
+    auto const get_edge_info = [&](EdgeID edge_id, NodeID node, auto const &edgeData) -> EdgeInfo {
         /// @todo Make string normalization/lowercase/trim for comparison ...
 
         auto const id = annotation[edgeData.annotation_data].name_id;
         BOOST_ASSERT(id != INVALID_NAMEID);
         auto const name = names.GetNameForID(id);
 
-        return {node,
+        return {edge_id,
+                node,
                 name,
+                edgeData.reversed,
                 edgeData.reversed ? 1 : 0,
                 annotation[edgeData.annotation_data].classes,
-                edgeData.flags.road_classification.GetClass()};
+                edgeData.flags};
+    };
+
+    auto is_bidirectional = [](auto flags) { return flags.is_split || (!flags.is_split && flags.forward && flags.backward); };
+
+    auto isSegregated = [&](NodeID node1,
+                            std::vector<EdgeInfo> v1,
+                            std::vector<EdgeInfo> v2,
+                            EdgeInfo const &current,
+                            double edgeLength) {
+        // Internal intersection edges must be short and cannot be a roundabout
+        // TODO adjust length as needed with lamda
+        if (edgeLength > 32.0f || current.flags.roundabout) {
+            return false;
+        }
+
+        // Print information about angles at node 1 from all inbound edge to the current edge
+        bool oneway_inbound = false;
+        for (auto const &edge_from : v1) {
+            // Get the inbound edge and edge data
+            auto edge_inbound = graph.FindEdge(edge_from.node, node1);
+            auto const &edge_inbound_data = graph.GetEdgeData(edge_inbound);
+            if (!edge_inbound_data.reversed) {
+                // Skip any inbound edges not oneway (i.e. skip bidirectional)
+                // and link edge
+                // and not a road
+                if (is_bidirectional(edge_inbound_data.flags)
+                        || edge_inbound_data.flags.road_classification.IsLinkClass()
+                        || (edge_inbound_data.flags.road_classification.GetClass() > RoadPriorityClass::SIDE_RESIDENTIAL)) {
+                    continue;
+                }
+
+                // Get the turn degree from the inbound edge to the current edge
+                auto const turn_degree = get_angle(edge_from.node, edge_inbound, current.edge);
+                // Skip if the inbound edge is not somewhat perpendicular to the current edge
+                // TODO add turn degree lamda
+                if (turn_degree > 150 && turn_degree < 210) {
```

Comment:
I think there is a helper function for determining if a turn is a straight angle somewhere. It might make sense to use it here.